### PR TITLE
[WIP] Adding core5/more core components

### PR DESCRIPTION
**Pull request by TingALin**
Adds more core components wrapper comprising of the following wrappers tag styling for apps:

- PageTitle
- MenusPadding
- PageRow
- RowPrimary
- RowSecondary
- Separator
- TopBar
- ButtonRow

### CODE REVIEW

File: styles/core.scss [hunk 1]
```diff
@@ -137,3 +119,304 @@ SPDX-License-Identifier: Apache-2.0 */
...
+  /* kill heading padding, already applied to wrapper */
+  h1,
+  h2,
+  h3,
+  h4 {
+    margin-top: 0;
+  }
+}
+
+.core-page-title {
```
Comment (rossbulat) [inline styles/core.scss:193]: It looks like this CSS has just been copied and pasted from the staking dashboard with no real thought of how to optimise it in this repo. For example, the header tags above should probably have 0 margin by default, set in Entry, so we don't have to keep resetting it.

I think a lot of work needs to be done to this section of the PR.

File: styles/_variables.scss [hunk 2]
```diff
@@ -1,10 +1,9 @@
 /* Copyright 2023 @paritytech/polkadot-dashboard-ui authors & contributors
 SPDX-License-Identifier: Apache-2.0 */
 
-
 $page-width-small-threshold: 850px;
 $page-width-medium-threshold: 1175px;
+$row-width-full-threshold: 1050px;
```
Comment (rossbulat) [inline styles/_variables.scss:6]: this variable name is ambiguous and it is not obvious what it is being applied to.

File: lib/core/types.ts [hunk 3]
```diff
@@ -16,3 +16,20 @@ export type SideProps = ComponentBase & {
   // whether side menu is in minimised state
   minimised: boolean;
 };
+
+export type PageTitleProps = ComponentBase & {
+  sticky?: boolean;
+};
+
+export type PageRowProps = ComponentBase & {
+  noVerticalSpacer?: boolean;
```
Comment (rossbulat) [inline lib/core/types.ts:25]: Comments and naming optimisations can probably be done here.

File: lib/core/index.tsx [hunk 4]
```diff
@@ -78,3 +86,126 @@ export const Side = ({ children, style, open, minimised }: SideProps) => (
...
+  <div className="core-separator" style={style}>
+    {children}
+  </div>
+);
+
+/* TopBar
+ *
+ * Positioned under titles for a Go Back button and other page header info.
+ */
+export const TopBar = ({ children, style }: ComponentBase) => (
```
Comment (rossbulat) [inline lib/core/index.tsx:187]: better name is warranted.

File: lib/core/index.tsx [hunk 5]
```diff
@@ -78,3 +86,126 @@ export const Side = ({ children, style, open, minimised }: SideProps) => (
     {children}
   </div>
 );
+
+/* PageTitle
+ *
+ * The element that wraps a page title. Determines the padding and position relative to top of screen when the element is stuck.
+ */
+export const PagesTitle = forwardRef(
```
Comment (rossbulat) [inline lib/core/index.tsx:94]: PageTitle

**Review by rossbulat — CHANGES_REQUESTED** [review]
Very much a draft, needs work.

File: styles/core.scss (hunk 1 above)
Comment (TingALin) [reply styles/core.scss:193]: understood, I will review them all

**Comment by rossbulat** [comment]
let's start this again with one component at a time.

**Pull request closed without merging**